src/frequenz/client/common/microgrid/__init__.py

```diff
@@ -1,4 +1,51 @@
 # License: MIT
-# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
 
-"""Frequenz microgrid definition."""
+"""Module to define the microgrid used with the common client."""
+
+from __future__ import annotations  # required for constructor type hinting
+
+from dataclasses import dataclass
+from typing import Self
+
+# pylint: disable=no-name-in-module
+from frequenz.api.common.v1.microgrid.microgrid_pb2 import (
+    MicrogridComponentIDs as PBMicrogridComponentIDs,
+)
+# pylint: enable=no-name-in-module
+
+@dataclass(frozen=True, kw_only=True)
+class MicrogridComponentIDs:
+    """Linking component IDs with their respective microgrid ID."""
+
+    microgrid_id: int
+    """The ID of the microgrid."""
+
+    component_ids: Sequence[int]
```

Comment:
`Sequence` was not imported

Reply:
Actually the type here might depend on the resolution of https://github.com/frequenz-floss/frequenz-client-common-python/pull/14/files#r1495647444